pkg/catalog/catalog.go

```diff
@@ -129,8 +131,67 @@ func getDefaultBuilderImages(client *occlient.Client) ([]CatalogImage, error) {
 		imageStreamTagMap[imageStreamTag.Name] = imageStreamTag
 	}
 
-	var builderImages []CatalogImage
+	builderImages := getBuildersFromImageStreams(imageStreams, imageStreamTagMap)
+
+	return builderImages, nil
+}
 
+// SpliceSupportedTags splits the tags in to fully supported and unsupported tags
+func SpliceSupportedTags(catalogImage CatalogImage) ([]string, []string) {
+
+	var supTag, nonSupTag []string
+	tagMap := createImageMap(catalogImage.imageStreamRef.Spec.Tags)
+	for _, tag := range catalogImage.NonHiddenTags {
+		imageName := tagMap[tag]
+		if isSupportedImage(imageName) {
+			supTag = append(supTag, tag)
+		} else {
+			nonSupTag = append(nonSupTag, tag)
+		}
+	}
+	return supTag, nonSupTag
+}
+
+func createImageMap(tagRefs []imagev1.TagReference) map[string]string {
+	tagMap := make(map[string]string)
+	for _, tagRef := range tagRefs {
+		// TODO: the From here can be of multiple types so we need to handle that
+		imageName := tagRef.From.Name
+		if tagRef.From.Kind == "DockerImage" {
+			urlImageName := strings.SplitN(tagRef.From.Name, "/", 2)[1]
+			ns, img, _, _, _ := occlient.ParseImageName(urlImageName)
+			imageName = ns + "/" + img
```

Comment:
Could using `imageName` as a variable here and four lines above open up doors to ambiguity in future? Should we name these differently to avoid any confusion?

Reply:
actually this is also part of the logic, that `imageName` by default be `tagRef.From.Name`, if its a dockerImage then it should be `ns/img` and if its imagestreamtag then we should look it up from existing map we made so far eg tag `latest` will point to an existing tag called `1.6` which should already be present in the map